fetch/data-urls/resources/data-urls.json

```diff
@@ -0,0 +1,175 @@
+  ["data:text/plain ,X",
+   "text/plain",
+   [88]],
+  ["data:text/plain%20,X",
+   "text/plain%20",
+   [88]],
+  ["data:text/plain\f,X",
+   "text/plain%0C",
+   [88]],
+  ["data:text/plain%0C,X",
+   "text/plain%0C",
+   [88]],
+  ["data:text/plain;,X",
+   "text/plain",
+   [88]],
+  ["data:;x=x;charset=x,X",
+   "text/plain;x=x;charset=x",
+   [88]],
+  ["data:;x=x,X",
+   "text/plain;x=x",
+   [88]],
+  ["data:text/plain;charset=windows-1252,%C2%B1",
+   "text/plain;charset=windows-1252",
+   [194, 177]],
+  ["data:text/plain;Charset=UTF-8,%C2%B1",
+   "text/plain;Charset=UTF-8",
+   [194, 177]],
+  ["data:image/gif,%C2%B1",
+   "image/gif",
+   [194, 177]],
+  ["data:IMAGE/gif,%C2%B1",
+   "IMAGE/gif",
+   [194, 177]],
+  ["data:IMAGE/gif;hi=x,%C2%B1",
+   "IMAGE/gif;hi=x",
+   [194, 177]],
+  ["data:IMAGE/gif;CHARSET=x,%C2%B1",
+   "IMAGE/gif;CHARSET=x",
+   [194, 177]],
+  ["data: ,%FF",
+   "",
+   [255]],
+  ["data:%20,%FF",
+   "",
+   [255]],
+  ["data:\f,%FF",
+   "",
+   [255]],
+  ["data:%1F,%FF",
+   "",
+   [255]],
+  ["data:\u0000,%FF",
+   "",
+   [255]],
+  ["data:%00,%FF",
+   "",
+   [255]],
+  ["data:text/html  ,X",
+   "text/html",
+   [88]],
+  ["data:†,X",
+   "",
+   [88]],
+  ["data:†/†,X",
+   "%E2%80%A0/%E2%80%A0",
+   [88]],
+  ["data:X,X",
+   "",
+   [88]],
+  ["data:image/png,X X",
+   "image/png",
+   [88, 32, 88]],
+  ["data:application/xml,X X",
+   "application/xml",
+   [88, 32, 88]],
+  ["data:unknown/unknown,X X",
+   "unknown/unknown",
+   [88, 32, 88]],
+  ["data:text/plain;a=\",\",X",
+   "data:text/plain;a=\"",
+   [34, 44, 88]],
+  ["data:text/plain;a=%2C,X",
+   "data:text/plain;a=%2C",
+   [88]],
+  ["data:;base64;base64,WA",
+   "text/plain;base64",
+   [88]],
+  ["data:x/x;base64;base64,WA",
+   "x/x;base64",
+   [88]],
+  ["data:x/x;base64;charset=x;base64,WA",
+   "x/x;base64;charset=x",
+   [88]],
+  ["data:x/x;base64;base64x,WA",
+   "x/x;base64;base64x",
+   [87, 65]],
+  ["data:;base64,W%20A",
+   "",
+   [88]],
+  ["data:;base64,W%0CA",
+   "",
+   [88]],
+  ["data:x;base64x,WA",
+   "",
+   [87, 65]],
+  ["data:x;base64;x,WA",
+   "",
+   [88]],
+  ["data:x;base64=x,WA",
+   "",
+   [87, 65]],
+  ["data:; base64,WA",
+   "",
+   [87, 65]],
+  ["data:;base64;,WA",
+   "",
+   [87, 65]],
+  ["data:;base64 ,WA",
+   "",
+   [88]],
+  ["data:;base 64,WA",
+   "",
+   [87, 65]],
+  ["data:;BASe64,WA",
+   "",
+   [88]],
+  ["data:;%62ase64,WA",
+   "",
+   [87, 65]],
+  ["data:%3Bbase64,WA",
+   "",
+   [87, 65]],
+  ["data:;charset=x,X",
+   "text/plain;charset=x",
+   [88]],
+  ["data:; charset=x,X",
+   "text/plain;charset=x",
+   [88]],
+  ["data:;charset =x,X",
+   "text/plain;charset=x",
+   [88]],
+  ["data:;charset= x,X",
+   "text/plain;charset=x",
+   [88]],
+  ["data:;charset=,X",
+   "text/plain;charset",
+   [88]],
+  ["data:;charset,X",
+   "text/plain;charset",
+   [88]],
+  ["data:;charset=\"x\",X",
+   "text/plain;charset=\"x\"",
+   [88]],
+  ["data:;CHARSET=\"X\",X",
```

Comment:
I guess this would be the place to test the questions in https://github.com/whatwg/fetch/pull/579#issuecomment-330542244? Or are such tests already here?

Reply:
Yeah, a little above this one. Line 163, 166, and various tests around those too.